File: leetcode/easy/ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0:
            return 1
        ways = [1, 1]
        for i in range(2, n + 1):
            ways.append(ways[i - 1] + ways[i - 2])
        return ways[n]

File: leetcode/easy/ClimbingStairs.java
class Solution {
    public int climbStairs(int n) {
        if (n == 0) {
            return 1;
        }
        int[] ways = new int[n + 1];
        ways[0] = ways[1] = 1;
        for (int i = 2; i <= n; i++) {
            ways[i] = ways[i - 1] + ways[i - 2];
        }
        return ways[n];
    }
}
